lms/djangoapps/courseware/access.py

```diff
@@ -687,3 +688,11 @@ def get_user_role(user, course_key):
         return 'staff'
     else:
         return 'student'
+
+
+def in_preview_mode():
+    """
+    Returns whether the user in preview mode or not.
+    """
+    hostname = get_current_request_hostname()
+    return hostname and settings.PREVIEW_DOMAIN in hostname.split('.')
```

Comment:
What happens on devstack when there isn't a separate domain? Do some of the community run both Studio and LMS on the same domain?